### ISSUE TITLE
[Bug]: Install Static Route with two % since 10.2

### ISSUE THREAD

**Issue by dpfaffenbauer** [pimcore/pimcore#10608]
### Expected behavior

Static Routes containing two % sign that are installed via API don't work anymore. The Problem is that % is a reserved "keyword" for resolving dynamic parameters like env vars.  Installed route should look like (note that % is escaped with another %):

```yaml
pimcore:
    staticroutes:
        definitions:
            0d275fff-e1cc-4f46-8fc1-ee9544065dd5:
                name: coreshop_partial_cart_add
                pattern: '/(\w+)\/_partial\/cart\/add\/([0-9]+)$/'
                reverse: '/%%_locale/_partial/cart/add/%%product'
                controller: 'CoreShop\Bundle\FrontendBundle\Controller\CartController:addItemAction'
                variables: '_locale,product'
                defaults: null
                siteId: null
                methods: {  }
                priority: 3
                creationDate: 1634228168
                modificationDate: 1634228168
```

Resulting in following error:

```
Symfony\Component\DependencyInjection\Exception\ParameterNotFoundException^ {#5561
  -key: "_locale/shop/customer/addresses/delete/"
  -sourceId: null
  -sourceKey: null
  -alternatives: []
  -nonNestedAlternative: null
  #message: "You have requested a non-existent parameter "_locale/shop/customer/addresses/delete/"."
  #code: 0
  #file: "./vendor/symfony/dependency-injection/ParameterBag/ParameterBag.php"
  #line: 93
  trace: {
    ./vendor/symfony/dependency-injection/ParameterBag/ParameterBag.php:93 { …}
    ./vendor/symfony/dependency-injection/ParameterBag/EnvPlaceholderParameterBag.php:61 { …}
    ./vendor/symfony/dependency-injection/ParameterBag/ParameterBag.php:217 { …}
    Symfony\Component\DependencyInjection\ParameterBag\ParameterBag->Symfony\Component\DependencyInjection\ParameterBag\{closure}() {}
    ./vendor/symfony/dependency-injection/ParameterBag/ParameterBag.php:206 { …}
    ./vendor/symfony/dependency-injection/ParameterBag/ParameterBag.php:175 { …}
    ./vendor/symfony/dependency-injection/ParameterBag/ParameterBag.php:165 { …}
    ./vendor/symfony/dependency-injection/ParameterBag/ParameterBag.php:165 { …}
    ./vendor/symfony/dependency-injection/ParameterBag/ParameterBag.php:165 { …}
    ./vendor/symfony/dependency-injection/ParameterBag/ParameterBag.php:165 { …}
    ./vendor/symfony/dependency-injection/ParameterBag/ParameterBag.php:165 { …}
    ./vendor/symfony/dependency-injection/Compiler/MergeExtensionConfigurationPass.php:62 { …}
    ./vendor/symfony/http-kernel/DependencyInjection/MergeExtensionConfigurationPass.php:39 { …}
    ./vendor/symfony/dependency-injection/Compiler/Compiler.php:91 { …}
    ./vendor/symfony/dependency-injection/ContainerBuilder.php:744 { …}
    ./vendor/symfony/http-kernel/Kernel.php:545 { …}
    ./vendor/pimcore/pimcore/lib/Kernel.php:237 { …}
    ./vendor/symfony/http-kernel/Kernel.php:786 { …}
    ./vendor/symfony/http-kernel/Kernel.php:125 { …}
    ./vendor/pimcore/pimcore/lib/Kernel.php:216 { …}
    ./src/Kernel.php:37 {
      Kernel->boot()^
      › 
      ›     parent::boot();
      › }
    }
    ./vendor/pimcore/pimcore/lib/Bootstrap.php:255 { …}
    ./vendor/pimcore/pimcore/lib/Bootstrap.php:60 { …}
    ./bin/console:44 { …}
  }
}

```

### Actual behavior

```yaml
pimcore:
    staticroutes:
        definitions:
            0d275fff-e1cc-4f46-8fc1-ee9544065dd5:
                name: coreshop_partial_cart_add
                pattern: '/(\w+)\/_partial\/cart\/add\/([0-9]+)$/'
                reverse: '/%_locale/_partial/cart/add/%product'
                controller: 'CoreShop\Bundle\FrontendBundle\Controller\CartController:addItemAction'
                variables: '_locale,product'
                defaults: null
                siteId: null
                methods: {  }
                priority: 3
                creationDate: 1634228168
                modificationDate: 1634228168

```

### Steps to reproduce

```php
$route = Staticroute::create();
$route->setName('coreshop_cart_create_qoute');
$route->setPattern('/(\w+)\/shop\/cart\/quote$/');
$route->setReverse('/%_locale/shop/cart/quote');
$route->setController('CoreShop\Bundle\FrontendBundle\Controller\CartController:createQuoteAction');
$route->setVariables('_locale');
$route->setPriority(3);
$route->save();
```

**Comment by dpfaffenbauer** [comment]
Also I think that the definition name could and should be the route name. Route Names are unique anyway.... therefore it could be the identifier within the symfony config

**Comment by dvesh3** [comment]
@dpfaffenbauer sorry, not able to reproduce with steps provided:
```php
    $route = Staticroute::create();
    $route->setName('test_route');
    $route->setPattern('/(\w+)\/test\/link$/');
    $route->setReverse('/%_locale/test/link');
    $route->setController('App\Controller\DefaultController:defaultAction');
    $route->setVariables('_locale');
    $route->setPriority(3);
    $route->save();
```
![image](https://user-images.githubusercontent.com/5137917/137452861-668ad20b-b6e6-468c-aa65-c4ce9e7b8001.png)

![image](https://user-images.githubusercontent.com/5137917/137454086-7cf7cd0e-d837-40bd-88d0-64c342764ba0.png)

**Comment by solverat** [comment]
%_locale is recognized by symfony that's why your simple example is working. just checkout your own examples from the documentation, they won't work anymore: https://pimcore.com/docs/pimcore/current/Development_Documentation/MVC/Routing_and_URLs/Custom_Routes.html

**Comment by dpfaffenbauer** [comment]
@dvesh3 my example is not right. It happens when you have two % signs in the config like:

```
$route = Staticroute::create();
$route->setName('coreshop_partial_cart_add');
$route->setPattern('/(\w+)\/_partial\/cart\/add\/([0-9]+)$/');
$route->setReverse('/%_locale/_partial/cart/add/%product');
$route->setController('CoreShop\Bundle\FrontendBundle\Controller\CartController:addItemAction');
$route->setVariables('_locale,product');
$route->setPriority(3);
$route->save();
```

Everything between the % signs is tried to be matched as parameter.

**Comment by dvesh3** [comment]
@dpfaffenbauer I got the point now. thanks!

> Also I think that the definition name could and should be the route name. Route Names are unique anyway.... therefore it could be the identifier within the symfony config

There could be routes with same name for multiple sites, so route name is not unique in that case.

It is still possible to set Id from API e.g., `$route->setId('coreshop_partial_cart_add');`.

**Comment by brusch** [comment]
Regarding the definition name, actually we can't use the name here, as it is possible to use the same route name for different sites 😉

**Comment by dpfaffenbauer** [comment]
@dvesh3 @brusch didn't know that, thought they were unique. but good to know. Other than that, PR is done #10617

**Comment by dvesh3** [comment]
Fixed by #10617

**Comment by solverat** [comment]
In my tests, I cannot create and use static routes dynamically anymore. They also won't get deleted from the file system via functional tests. I need to do some more debugging though...
